assignment-2/lean.md

```diff
@@ -1,3 +1,147 @@
+  optimal one. It is solved by standardising the workflow optimally.
+- *Mura* is the inconsistency of the stock of products available. It
+  is solved by a JIT (Just In Time) inventory: there is only a little
+  or no inventory of products or components and when there is a
+  request of a product, the components are requested through a pull
+  system in which all the components are recursively requested until
+  they reach the external supplier.
+- *Muda* is the over waste in its more general meaning: all those
+  added processes or goods that are not request or have any value for
+  the customer. *Muda* can be divided into two types: *muda* type I,
+  which has no value for the customer but it is necessary for the
+  manufacturing of the product; and *muda* type 2, which has no value
+  for the customer and it is not necessary either. It aims to
+  eliminate the *muda* type 2 but there is no straightforward way of
+  doing it because of the variety of things that can be
+  considered waste.
+
+The concept of waste is a traversal concept in other software
+development methodologies. *RUP* (Rational Unified Process) focuses on
+reducing risks and scrum focuses on delivering product. Both of them do
+not pay special attention to waste. Although scrum might look like aims
+to reduce waste as they only focus on delivering a product, their focus
+is to deliver whichever is the cost. This may lead to suboptimal
+practices.
+
+On one hand, RUP can be more wasteful than scrum. The early stages of
+RUP (business modelling, requirements and analysis & design) take a long
+period and effort and, although they produce deliverables and artefacts,
+they are not the requested one by the client. This qualifies as *muda*
+type I, as it is a waste, but it is necessary. This waste does not exist
+in scrum. In scrum, the documentation is almost not existing or it is
+very reduced based in the Agile Manifesto directives \[2\]. This means
+that the team does not produce any other deliverables but the product
+that the client requests.
+
+RUP also has more phases besides development. The early phases focus on
+having clear requirements and late ones focus on deployment and testing.
+These phases are also present in scrum but they happen in a different
+way. In scrum, part of the requirements are gathered by the product
+owner from the clients and the other part is done by the developers
+during the sprints. The testing and deployment are also integrated
+during the sprint instead in a separate phase. In RUP, the division of
+the phases aims to reduce the risks in the project by specifying the
+details of the project ahead and making sure that the product has been
+properly deployed and tested. Although necessary, these phases add extra
+waste in terms of processes and management, instead of being integrated
+within the development. In scrum, this waste does not exist because it
+is considered part of the development and it does not carry extra
```

Comment:
Does it matter if it is in phases or not? Does that change the total amount of work?